src/coreclr/vm/assemblynative.cpp

```diff
@@ -1407,3 +1409,57 @@ void QCALLTYPE AssemblyNative::TraceSatelliteSubdirectoryPathProbed(LPCWSTR file
 
     END_QCALL;
 }
+
+// static
+INT32 QCALLTYPE AssemblyNative::ApplyHotReloadUpdate(
+    QCall::AssemblyHandle assembly,
+    UINT8* metadataDelta,
+    INT32 metadataDeltaLength,
+    UINT8* ilDelta,
+    INT32 ilDeltaLength,
+    UINT8* pdbDelta,
+    INT32 pdbDeltaLength)
+{
+    QCALL_CONTRACT;
+
+    INT32 result = E_NOTIMPL;
+
+    BEGIN_QCALL;
+
+    _ASSERTE(assembly != nullptr);
+    _ASSERTE(metadataDelta != nullptr);
+    _ASSERTE(metadataDeltaLength > 0);
+    _ASSERTE(ilDelta != nullptr);
+    _ASSERTE(ilDeltaLength > 0);
+
+#ifdef EnC_SUPPORTED
+    GCX_COOP();
+    {
+        if (!CORDebuggerAttached())
```

Comment:
Why is this blocked when the debugger is attached? It should work just fine when the debugger is attached.

I understand that there is a problem with both debugger and this API apply conflicting deltas. We should detect that situation and only report error once that happens.

Reply:
Also, blocking this API when the managed debugger is attached will make it really hard for engineers outside the core runtime team to debug any hot-reload related issues.

Reply:
I wanted to keep things simple and the current thoughts on the overall user experience does need for debugging and hot reload to work at the same time.  We want to (eventually) smoothly transition from hot reload changes to debugging (which is going to take work to update the debugger and DAC/DBI metadata copies). 

One of the biggest problems with allowing it is that debugger messages are sent to the right side when a function or field is added and that assumes the process is suspended/stopped in the debugger.

Reply:
Can we just not let debugger know about any edits made using this API, for now at least?

Reply:
I can add a "updateDebugger" flag to EditAndContinueModule::ApplyEditAndContinue and use that to control the debugger messages.

Reply:
I think it would be useful to make the hot reload e2e easier to debug.

Reply:
I added some simple invalid parameter testing for the new API in the latest commit.